src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskCancelByIdControllerTest.java

```diff
@@ -115,11 +130,114 @@ void should_succeed_and_return_204_and_update_cft_task_state() throws Exception
+
+        Optional<TaskResource> taskResource = cftTaskDatabaseService.findByIdOnly(taskId);
+
+        assertTrue(taskResource.isPresent());
+        assertEquals(CANCELLED, taskResource.get().getState());
+    }
+
+    @Test
+    void should_not_cancel_and_return_403_wih_no_cancel_permission() throws Exception {
+
+        mockServices.mockUserInfo();
+        List<RoleAssignment> roleAssignments = new ArrayList<>();
+
+        RoleAssignmentRequest roleAssignmentRequest = RoleAssignmentRequest.builder()
+            .testRolesWithGrantType(TestRolesWithGrantType.STANDARD_TRIBUNAL_CASE_WORKER_PUBLIC)
+            .roleAssignmentAttribute(
+                RoleAssignmentAttribute.builder()
+                    .jurisdiction("IA")
+                    .caseType("Asylum")
+                    .caseId("cancelCaseSuccessId1")
+                    .build()
+            )
+            .build();
+
+        createRoleAssignment(roleAssignments, roleAssignmentRequest);
+        RoleAssignmentResource accessControlResponse = new RoleAssignmentResource(roleAssignments);
+        TaskRoleResource tribunalResource = new TaskRoleResource(
+            "tribunal-caseworker", true, false, false, false, false,
+            true, new String[]{}, 1, false, "LegalOperations",
+            taskId, OffsetDateTime.now(), false, false, false, true,
+            false, false, false, false, false, false
+        );
+        insertDummyTaskInDb(taskId, tribunalResource);
+
+        when(accessControlService.getRoles(IDAM_AUTHORIZATION_TOKEN))
+            .thenReturn(new AccessControlResponse(mockedUserInfo, roleAssignments));
+
+        when(idamWebApi.token(any())).thenReturn(new Token(IDAM_AUTHORIZATION_TOKEN, "scope"));
+        when(serviceAuthorisationApi.serviceToken(any())).thenReturn(SERVICE_AUTHORIZATION_TOKEN);
+
+        doNothing().when(camundaServiceApi).bpmnEscalation(any(), any(), any());
+
+        mockMvc.perform(
+                post(ENDPOINT_BEING_TESTED)
+                    .header(AUTHORIZATION, IDAM_AUTHORIZATION_TOKEN)
+                    .header(SERVICE_AUTHORIZATION, SERVICE_AUTHORIZATION_TOKEN)
+                    .contentType(MediaType.APPLICATION_JSON_VALUE)
+            )
+            .andExpectAll(
+                status().is4xxClientError(),
+                content().contentType(APPLICATION_PROBLEM_JSON_VALUE),
+                jsonPath("$.type").value("https://github.com/hmcts/wa-task-management-api/problem/role-assignment-verification-failure"),
+                jsonPath("$.title").value("Role Assignment Verification"),
+                jsonPath("$.status").value(403),
+                jsonPath("$.detail").value(
+                    "Role Assignment Verification: "
+                        + "The request failed the Role Assignment checks performed.")
+            );
+
+    }
+
+    @Test
+    void should_succeed_and_return_204_and_update_cft_task_state_with_cancelown_granular_permission() throws Exception {
```

Comment:
we agreed to remove this as it's a duplicate right?

Reply:
removed